velox/substrait/VariantToVectorConverter.cpp

```diff
@@ -62,6 +62,24 @@ VectorPtr setVectorFromVariantsByKind<TypeKind::VARCHAR>(
   }
   return flatVector;
 }
+
+template <>
+VectorPtr setVectorFromVariantsByKind<TypeKind::DATE>(
+    const std::vector<velox::variant>& values,
+    memory::MemoryPool* pool) {
+  auto flatVector =
+      BaseVector::create<FlatVector<int32_t>>(DATE(), values.size(), pool);
```

Comment:
int32_t -> Date

Reply:
@mbasmanova I attempted this, but it causes the following issue

```bash
VeloxSubstraitRoundTripTest.date1
unknown file: Failure
C++ exception with description "wrong kind! INTEGER != DATE" thrown in the test body.
```

This was my doubt in this PR as noted in the checklist. Do you think I have to do another modification?

Reply:
Can you debug this?

Reply:
Yes, I am on it now.

Reply:
The issue was that without the template Specialization piece this would give the same error as before. 

```C++
template <>
VectorPtr setVectorFromVariantsByKind<TypeKind::DATE>(
    const std::vector<velox::variant>& values,
    memory::MemoryPool* pool) {
  auto flatVector =
      BaseVector::create<FlatVector<Date>>(DATE(), values.size(), pool);

  for (vector_size_t i = 0; i < values.size(); i++) {
    if (values[i].isNull()) {
      flatVector->setNull(i, true);
    } else {
      flatVector->set(i, Date(values[i].value<int32_t>()));
    }
  }
  return flatVector;
}
```

With this change it works fine, although it is not pretty neat. The variant value set with Date seems to be `int32_t`. Not sure if my change is not correct when I put `Date` instead of `int32_t`.

Reply:
`values[i]` here is always an `INTEGER` so, if we use `Date` it cannot output the expected value. Either we have to wrap it back to `Date` or use the previous. I think there has to be modifications that need to be done to `variant` interface as far as I understand. But appreciate your thoughts.

Reply:
@mbasmanova what do you think about this?

Reply:
@vibhatha It looks like we need to fix conversion of Date constants in SubstraitVeloxExprConverter::toVeloxExpr:

```
    case ::substrait::Expression_Literal::LiteralTypeCase::kDate:
      return std::make_shared<core::ConstantTypedExpr>(
          variant(substraitLit.date()));
```

->

```
    case ::substrait::Expression_Literal::LiteralTypeCase::kDate:
      return std::make_shared<core::ConstantTypedExpr>(
          variant(Date(substraitLit.date())));
```

Reply:
I applied this one.